File: src/ipc/localfs.ts
import { invoke } from "@tauri-apps/api/core";

export interface LocalEntry {
  name: string;
  is_dir: boolean;
  size: number | null;
}

export const homeDir = (): Promise<string | null> => invoke("local_home_dir");
export const listDir = (path: string): Promise<LocalEntry[]> => invoke("local_list_dir", { path });

File: src-tauri/src/commands/localfs.rs
use crate::core::localfs::{self, LocalEntry};

#[tauri::command]
pub fn local_home_dir() -> Option<String> {
    localfs::home_dir()
}

#[tauri::command]
pub fn local_list_dir(path: String) -> Result<Vec<LocalEntry>, String> {
    localfs::list_dir(&path).map_err(|e| e.to_string())
}
